src/languages.ts

```diff
@@ -67,12 +90,17 @@ export const languages: {
             "try",
             "quote",
             "with",
+            "defprotocol",
+            "defimpl",
         ],
         closeTokens: [
             "end",
+            ", do"
         ],
         neutralTokens: [
             "do",
+            " -> ",
+            " <- ",
```

Comment:
I also had problems making this work. Maybe it's due to the way the Regexp is matched?